framework/db/ActiveRecord.php

```diff
@@ -115,9 +115,11 @@ class ActiveRecord extends BaseActiveRecord
      */
     public function loadDefaultValues($skipIfSet = true)
     {
-        foreach (static::getTableSchema()->columns as $column) {
-            if ($column->defaultValue !== null && (!$skipIfSet || $this->{$column->name} === null)) {
-                $this->{$column->name} = $column->defaultValue;
+        $columns = static::getTableSchema()->columns;
+        foreach ($this->attributes() as $name) {
+            $defaultValue = $columns[$name]->defaultValue;
```

Comment:
It could be that `attributes()` contain an attribute name that is not in the table schema. If it's true, the code will error.

Reply:
@samdark I don't think
https://github.com/yiisoft/yii2/blob/4c1f872320310ad719e660f183c5dc91088463b1/framework/db/BaseActiveRecord.php#L1072

Reply:
@samdark  fixed

Reply:
@samdark but if attribute not in columns then user must override update/insert methods